python/ray/train/torch.py

```diff
@@ -190,7 +193,7 @@ def __iter__(self):
             yield from iterator
 
         for item in iterator:
-            yield (i.to(self.device) for i in item)
+            yield self._move_to_device(item)
```

Comment:
Also, shouldn't this be in an `else` block, otherwise the second yield will execute after the first?

Reply:
Ah yeah good point. Functionally, I think the current code is still correct since once you finish `yield from iterator`, the following for loop won't be entered since the iterator is exhausted. But I will add an `else` block for readability.

Reply:
Updated!